lib/api/core/plugins/pluginContext.js

```diff
@@ -252,4 +262,50 @@ function instantiateRequest(request, data, options = {}) {
   return target;
 }
 
+/**
+ * Returns a currified function of pluginsManager.registerStrategy
+ *
+ * @param  {Kuzzle} kuzzle 
+ * @param  {string} pluginName 
+ * @return {function} function taking a strategy name and properties, 
+ *                    registering it into kuzzle, and returning
+ *                    a promise
+ */
+function curryAddStrategy(kuzzle, pluginName) {
+  return function addStrategy(name, strategy) {
+    return new Bluebird((resolve, reject) => {
+      try {
+        kuzzle.pluginsManager.registerStrategy(pluginName, name, strategy);
```

Comment:
Didn't like the name you proposed but apart from that: events added.

I'm working on the cluster code to make nodes synchronize dynamic strategies management.